rest/config.go

```diff
@@ -719,8 +719,19 @@ func (config *ServerConfig) deprecatedConfigLoggingFallback() (warnings []base.D
 			warnings = append(warnings, func() {
 				base.Warnf(warningMsgFmt, `logging.["default"].LogFilePath`, "logging.log_file_path")
 			})
+
 			// Set the new LogFilePath to be the directory containing the old logfile, instead of the full path.
-			config.Logging.LogFilePath = filepath.Dir(*config.Logging.DeprecatedDefaultLog.LogFilePath)
+			// SGCollect relies on this path to pick up the standard and rotated log files.
+			info, err := os.Stat(*config.Logging.DeprecatedDefaultLog.LogFilePath)
+			if err != nil {
+				base.Infof(base.KeyAll, "Detected error retrieving stats of logging.[\"default\"].LogFilePath %q, "+
+					"Error: %v", *config.Logging.DeprecatedDefaultLog.LogFilePath, err)
+			}
+			if !os.IsNotExist(err) && info.IsDir() {
+				config.Logging.LogFilePath = *config.Logging.DeprecatedDefaultLog.LogFilePath
+			} else {
```

Comment:
We're still falling into the first case when we get an error back from stat.
It sounds like what we need based on previous comments is `if err == nil && info.IsDir()` ?

https://github.com/couchbase/sync_gateway/pull/4704/files#r472316408

> As discussed, let's simplify to the following:
> 
>     * if config.Logging.DeprecatedDefaultLog.LogFilePath exists and is a directory, use it as the directory
> 
>     * in all other cases (it's a file, or it doesn't exist) use filepath.Dir to get parent directory

Reply:
Make sense, this is revised in latest revision.